File: src/common/py/core/messaging/dispatchers/event_dispatcher.py
from .message_dispatcher import MessageDispatcher
from ..event import Event
from ..meta import EventMetaInformation


class EventDispatcher(MessageDispatcher[Event]):
    # pylint: disable=protected-access
    """
    Message dispatcher specific to ``Event``.
    """

    def __init__(self):
        super().__init__(EventMetaInformation)

    def pre_dispatch(self, msg: Event, msg_meta: EventMetaInformation) -> None:
        """
        Adds command meta information to a global list so that command replies can be handled properly.

        Args:
            msg: The command that is about to be dispatched.
            msg_meta: The command meta information.
        """
        from ...logging import debug

        debug(f"Dispatching event: {msg}", scope="bus")
        super().pre_dispatch(msg, msg_meta)

File: src/common/web/core/messaging/dispatchers/EventDispatcher.ts
import logging from "../../logging/Logging";
import { Event } from "../Event";
import { EventMetaInformation } from "../meta/EventMetaInformation";
import { MessageDispatcher } from "./MessageDispatcher";

/**
 * Message dispatcher specific to ``Event``.
 */
export class EventDispatcher extends MessageDispatcher<Event, EventMetaInformation> {
    /**
     * Called to perform tasks *before* sending a message.
     *
     * This method is called before any service-registered message handler is invoked.
     *
     * @param msg - The message that is about to be dispatched.
     * @param msgMeta - The message meta information.
     *
     * @throws Error - If the meta information type is invalid.
     */
    public preDispatch<MsgType extends Event>(msg: MsgType, msgMeta: EventMetaInformation): void {
        logging.debug(`Dispatching event: ${String(msg)}`, "bus");
        super.preDispatch(msg, msgMeta);
    }
}
